staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -227,8 +269,13 @@ func (s *store) conditionalDelete(ctx context.Context, key string, out runtime.O
 			return err
```

Comment:
add a unit test to make sure the right thing happens if the suggestion is stale and the key no longer exists (was already deleted). I think we would get a NotFound here and return, but verify the right thing happens

Reply:
Added "TestDeleteWithSuggestionOfDeletedObject" test.

That said - it doesn't exercise this path. What happens is that the transation fails, we get into
```
if !txnResp.Succeeded {
```
branch and this is handled in getState (getResp contains empty KV field and that is handled as returning "not-found" error).

Reply:
`txnResp.Responses[0]` exists for a NotFound response?

Reply:
Yes - it exists and being GetResponse contains empty set of KV pairs.